contracts/mesh-provider/src/state.rs

```diff
@@ -51,6 +61,43 @@ impl Stake {
         val.shares_to_tokens(self.shares)
     }
 
+    /// Calculate rewards
+    pub fn calc_pending_rewards(
+        &mut self,
+        new_rptpb: Uint128,
+        staked: Uint128,
+        height: u64
+    ) -> Result<(), ContractError> {
+        if staked.is_zero() || self.rewards.last_height == 0 {
+            self.rewards.last_rptpb = new_rptpb;
+            self.rewards.last_height = height;
+            return Ok(());
+        }
+
+        let rptpb_to_pay = new_rptpb - self.rewards.last_rptpb;
+        let block_to_pay = height - self.rewards.last_height;
+
+        if rptpb_to_pay.is_zero() || block_to_pay == 0 {
+            self.rewards.last_rptpb = new_rptpb;
+            self.rewards.last_height = height;
+            return Err(ContractError::ZeroRewardsToSend {})
+        }
+
+        let pending_rewards = rptpb_to_pay
```

Comment:
This is gonna get huge.

rptpb is "rate per token per block"? It definitely needs a decimal there. As I expect much less than 1 token.

eg. ATOM with 11% APR on a validator with 10% commission, will pay out 0.1uatom on 1 atom staked **per year**

With 6 second blocks, this would be `0,000000019012853`which fits fine as a Decimal, but not as Uint128

Reply:
We also need to handle rounding issues here, maybe store pending as a Decimal now?

Reply:
Oh, wasn't sure what number I should expect honestly, so decimal for rptpb and pending rewards is the right way.